File: backend/src/main/java/com/bidbridge/backend/entity/LoadingPlace.java

package com.bidbridge.backend.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;
import java.time.Instant;

@Data
@NoArgsConstructor
@Entity
@Table(name = "loading_places", schema = "public")
public class LoadingPlace {

    @Id
    @GeneratedValue
    private UUID id;

    private String name;

    @ManyToOne
    @JoinColumn(name = "company_id")
    private Company company;

    private Instant createdAt;
}

File: frontend/src/types/loading-place.ts
export interface LoadingPlace {
  id: string;
  name: string;
  company_id: string;
  company_name: string;
  created_at: string;
}

export interface LoadingPlaceApiResponse {
  id: string;
  name: string;
  companyId: string;
  companyName: string;
  createdAt: string;
}

export function transformLoadingPlaceFromApi(
  apiLoadingPlace: LoadingPlaceApiResponse
): LoadingPlace {
  return {
    id: apiLoadingPlace.id,
    name: apiLoadingPlace.name,
    company_id: apiLoadingPlace.companyId,
    company_name: apiLoadingPlace.companyName,
    created_at: apiLoadingPlace.createdAt,
  };
}
